django/db/models/expressions.py

```diff
@@ -1142,6 +1142,10 @@ def __init__(self, alias, target, output_field=None):
         super().__init__(output_field=output_field)
         self.alias, self.target = alias, target
 
+    @cached_property
+    def identity(self):
+        return (self.__class__, self.alias, self.target, self.output_field)
```

Comment:
Had to move this back to a property as testing on django 4.2.2 showed no perf benefit otherwise anymore. I initially tested on 4.2.0

Reply:
That's suspicious. As far as I'm aware, we haven't done anything that could affect this optimization. Maybe it's not worth changing :thinking:

Reply:
Good catch :+1:

Reply:
I think it's no longer worth if it's only micro-optimization.